src/parser_urdf_TEST.cc

```diff
@@ -994,6 +1008,553 @@ TEST(URDFParser, ParseWhitespace)
+        <link name='link1'>
+          <inertial>
+            <mass value="1e-6" />
+            <origin rpy="1.570796326794895 0 0" xyz="0.123456789123456 0 0.0" />
+            <inertia ixx="0.01" ixy="0" ixz="0" iyy="0.01" iyz="0" izz="0.01" />
+          </inertial>
+        </link>
+      </robot>)";
+
+    sdf::URDF2SDF parser;
+    sdf::ParserConfig config_;
+    tinyxml2::XMLDocument sdfResult;
+    parser.InitModelString(str, config_, &sdfResult);
+
+    EXPECT_PRED2(contains, buffer.str(),
+        "urdf2sdf: link[link1] has a mass value of less than or equal to 1e-6, "
+        "not modeled in sdf");
+  }
+
+  {
+    // clear the contents of the buffer
+    buffer.str("");
+
+    std::string str = R"(
+      <robot name='test_robot'>
+        <link name='link1'>
+          <inertial>
+            <mass value="9e-7" />
+            <origin rpy="1.570796326794895 0 0" xyz="0.123456789123456 0 0.0" />
+            <inertia ixx="0.01" ixy="0" ixz="0" iyy="0.01" iyz="0" izz="0.01" />
+          </inertial>
+        </link>
+      </robot>)";
+
+    sdf::URDF2SDF parser;
+    sdf::ParserConfig config_;
+    tinyxml2::XMLDocument sdfResult;
+    parser.InitModelString(str, config_, &sdfResult);
+
+    EXPECT_PRED2(contains, buffer.str(),
+        "urdf2sdf: link[link1] has a mass value of less than or equal to 1e-6, "
+        "not modeled in sdf");
+  }
+
+  // Revert cerr rdbug so as to not interfere with other tests
+  std::cerr.rdbuf(old);
+  #ifdef _WIN32
+    sdf::Console::Instance()->SetQuiet(true);
+  #endif
+}
+
+/////////////////////////////////////////////////
+TEST(URDFParser, ConvertLinksWithNoInertiaToFrames)
+{
+  // Capture sdferr output
+  std::stringstream buffer;
+  auto old = std::cerr.rdbuf(buffer.rdbuf());
+
+  #ifdef _WIN32
+    sdf::Console::Instance()->SetQuiet(false);
+  #endif
+
+  {
+    // clear the contents of the buffer
+    buffer.str("");
+
+    std::string str = R"(
+      <robot name='test_robot'>
+        <link name='link1'/>
+      </robot>)";
+
+    sdf::URDF2SDF parser;
+    sdf::ParserConfig config_;
+    config_.URDFSetConvertLinkWithNoMassToFrame(true);
+    tinyxml2::XMLDocument sdfResult;
+    parser.InitModelString(str, config_, &sdfResult);
+
+    EXPECT_PRED2(contains, buffer.str(),
+        "urdf2sdf: link[link1] has no inertia defined, converting to a frame in"
+        " sdf");
+
+    tinyxml2::XMLElement *sdf = sdfResult.FirstChildElement("sdf");
+    ASSERT_NE(nullptr, sdf);
+    tinyxml2::XMLElement *model = sdf->FirstChildElement("model");
+    ASSERT_NE(nullptr, model);
+    tinyxml2::XMLElement *frame = model->FirstChildElement("frame");
+    ASSERT_NE(nullptr, frame);
+    ASSERT_NE(nullptr, frame->Attribute("name"));
+    EXPECT_EQ("link1", std::string(frame->Attribute("name")));
+    tinyxml2::XMLElement *link = model->FirstChildElement("link");
+    ASSERT_EQ(nullptr, link);
+  }
+
+  {
+    // clear the contents of the buffer
+    buffer.str("");
+
+    std::string str = R"(
```

Comment:
This example demonstrates the issue about assigning the correct `attached_to` frame. If we leave it unassigned, the frame `link1` will be rigidly attached to `link2` since `link2` will be the canonical link of the model. If we simulate this and actuate `joint1_2`, we'd expect the pose between `link1` and `link2` to change, but this won't be the case because they are rigidly attached. I don't think we can support such use cases.

Reply:
https://github.com/gazebosim/sdformat/pull/1238/commits/1e5a2da98984178ccf9812b0cc3c0c181c456ecf